src/Messages/OWSMessageSender.m

```diff
@@ -594,6 +594,19 @@ - (void)deliverMessage:(TSOutgoingMessage *)message
 {
     TSThread *thread = message.thread;
 
+#ifdef DEBUG
+    // Assert that we're only sending sync messages when necessary
+    if ([message isKindOfClass:[OWSOutgoingSyncMessage class]]) {
+        __block BOOL hasSecondaryDevices = NO;
+        [self.storageManager.dbConnection readWithBlock:^(YapDatabaseReadTransaction *_Nonnull transaction) {
```

Comment:
Here you use self.storageManager.dbConnection but below you use self.dbConnection.